pysat/tests/test_utils.py

```diff
@@ -466,77 +465,121 @@ def test_netcdf_attribute_override(self):
+            outfile, pandas_format=self.testInst.pandas_format)
 
         # Custom attribute correctly read from file
         assert meta.bespoke
+        return
 
 
-class TestBasicNetCDF4xarray():
-    """NOTE: combine with above class as part of #60"""
+class TestLoadNetCDF4XArray():
+    """Unit tests for `load_netcdf4` using xarray data.
+
+    Note
+    ----
+    Make this a TestLoadNetCDF4 class test as a part of fixing #60.
+
+    """
 
     def setup(self):
         """Runs before every method to create a clean testing setup."""
-        # store current pysat directory
+        # Store current pysat directory
         self.data_path = pysat.params['data_dirs']
 
-        # create temporary directory
+        # Create temporary directory
         self.tempdir = tempfile.TemporaryDirectory()
         pysat.params['data_dirs'] = [self.tempdir.name]
 
         self.testInst = pysat.Instrument(platform='pysat',
                                          name='testing2d_xarray',
-                                         update_files=True,
-                                         num_samples=100)
+                                         update_files=True, num_samples=100)
         self.stime = pysat.instruments.pysat_testing2d_xarray._test_dates[
             '']['']
 
-        # create testing directory
+        # Create testing directory
         prep_dir(self.testInst)
 
     def teardown(self):
         """Runs after every method to clean up previous testing."""
-        # remove_files(self.testInst)
+        # Reset the pysat parameters
         pysat.params['data_dirs'] = self.data_path
+
+        # Remove the temporary directory
         self.tempdir.cleanup()
-        del self.testInst, self.stime
+
+        # Clear the class attributes
+        del self.data_path, self.tempdir, self.testInst, self.stime
 
     def test_basic_write_and_read_netcdf4_default_format(self):
-        """ Test basic netCDF4 writing and reading
-        """
-        # create a bunch of files by year and doy
-        prep_dir(self.testInst)
+        """ Test basic netCDF4 writing and reading."""
+        # Write the output test data
         outfile = os.path.join(self.testInst.files.data_path,
                                'pysat_test_ncdf.nc')
         self.testInst.load(date=self.stime)
-        self.testInst.data.attrs['new_attr'] = 1
         self.testInst.data.to_netcdf(outfile)
 
-        loaded_inst, meta = pysat.utils.load_netcdf4(
+        # Load the written data
+        self.loaded_inst, meta = pysat.utils.load_netcdf4(
             outfile, pandas_format=self.testInst.pandas_format)
-        keys = self.testInst.data.data_vars.keys()
 
-        for key in keys:
-            assert(np.all(self.testInst[key] == loaded_inst[key]))
-        assert meta.new_attr == 1
+        # Compare the initial and loaded data
+        for key in self.testInst.data.data_vars.keys():
+            assert(np.all(self.testInst[key] == self.loaded_inst[key]))
+
```

Comment:
```suggestion
    del self.loaded_inst
```
Found it.  loaded xarray is not deleted first.  Tabs may need to be adjusted

Reply:
I updated setup/teardown instead.

Reply:
If that doesn't work I'll mess with the teardown order.

Reply:
It should probably be deleted before `tempdir.cleanup()` is invoked, since the open file link is what's breaking things on windows.

Reply:
I changed that in all of the netCDF test classes.  Fingers crossed! 🐦